python/sglang/srt/multimodal/processors/base_processor.py

```diff
@@ -233,18 +362,152 @@ def process_mm_data(
+                    raise ValueError("invalid img token num")
+
+                if self.hash_table.get(hash_key) is None:
+                    new_processed_img_idxes.append(img_idx)
+                    new_processed_imgs.append(images[img_idx])
+                else:
+                    remove_image_idx.append(img_idx)
+
+                img_heights.append(img_height)
+                img_token_nums.append(img_token_num)
+
+            processed_text = operate_substrings(
+                input_text, to_replace_str, remove_image_idx, repalce_str
+            )
+            kwargs["images"] = (
+                new_processed_imgs if len(new_processed_imgs) != 0 else None
+            )
+            result = processor.__call__(
+                text=[processed_text],
+                padding=True,
+                return_tensors="pt",
+                **kwargs,
+            )
+
+            for feature_name in self.FEATURE_NAMES:
+                if feature_name in result and isinstance(
+                    result[feature_name], torch.Tensor
+                ):
+                    result[feature_name] = result[feature_name].to("cpu")
+
+            start_height = 0
+            end_height = 0
+            tensor_lists = []
+            image_grid_thw_lists = []
+            use_cache_mark = []
+            delete_keys = []
+            for img_idx in range(len(images)):
+                # cache Tensor
+                if img_idx in new_processed_img_idxes:
+                    img_height = img_heights[img_idx]
+                    processed_img_heights.append(img_height)
+             
+                    start_height = end_height
+                    end_height = start_height + img_height
+                    to_cache_tensor = result["pixel_values"][start_height:end_height]
+                    delete_key = self.hash_table.add(
+                        img_hash_keys[img_idx], to_cache_tensor
+                    )
+                    if delete_key:
+                        delete_keys.append(delete_key)
+                    tensor_lists.append(to_cache_tensor)
+                    image_grid_thw_lists.append(
+                        [
+                            1,
+                            int(images[img_idx].size[1] // QWEN2_PATCH_SIZE),
+                            int(images[img_idx].size[0] // QWEN2_PATCH_SIZE),
+                        ]
+                    )
+                    use_cache_mark.append(False)
+                # add input ids and insert tensor
+                else:
+                    cached_tensor = self.hash_table.get(img_hash_keys[img_idx])
+                    assert isinstance(
+                        cached_tensor, torch.Tensor
+                    ), "invalid cached_tensor"
+                    # tensor_lists.append(cached_tensor)
+                    insert_cached_ids = (
+                        [v_start_token]
+                        + img_token_nums[img_idx] * [img_pad_token]
```

Comment:
I'm wondering if there is any `result['xx']` field that can be used directly to obtain img_token_num